src/test/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheckTest.java

```diff
@@ -138,6 +146,17 @@ public void testFileWithThreeTopLevelEnum() throws Exception {
         verify(checkConfig, getPath("InputOneTopLevelClassEnum2.java"), expected);
     }
 
+    @Test
+    public void testFileWithThreeTopLevelAnnotation() throws Exception {
```

Comment:
Please also add tests 
1. with annotations before annotation definitions (Retention policy, Target and other), like in report.
2. With mixed class / annotation / enum /interface (some existing test input ut can be extended)
3. I remember there is some special treat for public type definitions, it should be here as well

Reply:
Fixed the tests to cover the 3 points mentioned